bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -151,7 +166,20 @@ Java_com_bugsnag_android_ndk_NativeBridge_deliverReportAtPath(
   }
   remove(report_path);
   (*env)->ReleaseStringUTFChars(env, _report_path, report_path);
+  goto exit;
+
+  exit:
   pthread_mutex_unlock(&bsg_native_delivery_mutex);
+  if (jpayload != NULL) {
+    (*env)->ReleaseByteArrayElements(env, jpayload, (jbyte *)payload,
+                                     0); // <-- frees payload
```

Comment:
This has been addressed in the following PR for v5: https://github.com/bugsnag/bugsnag-android/pull/1142

There are still some remaining backports to be done from v5 that add the method used in the above changeset. Therefore I'd suggest that this should be addressed as part of a separate PR rather than part of this.

Reply:
👍